### ISSUE TITLE
Popup for keymap set by set-transient-map

### ISSUE THREAD

**Issue by noctuid** [justbur/emacs-which-key#164]
I'd like to be able to have which-key appear after a call to `set-transient-map` if possible. It would be great if this could be done automatically, but if there was some way to do this manually, that would be fine as well for my use case.

**Comment by justbur** [comment]
I can do it, but there are a couple of decisions to make

  1. Should all bindings be shown (since many more are accessible) or just the ones in the transient map? The transient ones could be sorted first or highlighted
  2. Is it important to have access to the page turning commands, like `C-h n`? At first glance, this looks tricky with transient maps.

**Comment by noctuid** [comment]
I'd only want the ones in the transient map to be shown. I'd think that almost always one of those keys would be pressed next. If you think it could be useful to show all the other available keys, then maybe it could be an option.

Could you elaborate on this? Is this related to 1 (do you mean the default `C-h n` or if the command was bound in the transient map)?

**Comment by justbur** [comment]
On 2, normally `C-h` gets diverted into a set of commands to change the which-key page, undo the last key in the sequence, etc. This is already a bit tricky to implement, and with a transient map it looks even more difficult due to the nature of transient maps. 

Also, I'm curious why a hydra wouldn't be better suited for this use case.

**Comment by noctuid** [comment]
If I'm understanding correctly, you mean that having any prefix keys in the keymap would be an issue? Ideally those would be accessible.

The idea would be to use it with already existing keymaps. Some people prefer to define hydras even when all the bindings exit immediately just for the popup text, but I'd rather just have the command names automatically shown with which-key.

The initial use case I had in mind only appliesl in a very specific, uncommon instance (where the choice currently is between showing a which-key popup and breaking macro recording, or at least requiring a more complicated workaround to get it to work properly). I could explain if you are curious, but I don't think it is a very convincing example because of how unlikely it is to be an issue. I did think that this could be useful in other situations as well though, which is why I made this issue. For example, consider evil-snipe, which uses `set-transient-map` after seeking to set the key used to seek and its counterpart (e.g. `f` and `F`)  to cycle like `;` and `,`. It could potentially be useful to visualize temporary changes like these.

If you don't think that this is a feature worth implementing due to the potential difficulty, then I think that is fine. It's a not at all a necessary feature I think.
